### PR TITLE
Feature/final versions

### PR DESCRIPTION
**Pull request by alexflav23**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -37,12 +37,12 @@ lazy val microservice = Project(appName, file("."))
...
-      "uk.gov.hmrc"       %% "simple-reactivemongo-26" % "0.5.0",
-      "uk.gov.hmrc"       %% "mongo-lock"              % "5.2.0-SNAPSHOT",
-      "uk.gov.hmrc"       %% "metrix"                  % "2.0.0-3-g48b8866",
+      "uk.gov.hmrc"       %% "simple-reactivemongo-26" % "0.8.0",
+      "uk.gov.hmrc"       %% "mongo-lock-26"              % "0.2.0",
+      "uk.gov.hmrc"       %% "metrix-26"                  % "0.3.0",
       "org.scalatest"     %% "scalatest"               % "2.2.6"             % Test,
       "org.pegdown"       % "pegdown"                  % "1.6.0"             % Test,
-      "uk.gov.hmrc"       %% "reactivemongo-test"      % "3.0.0"             % Test,
+      "uk.gov.hmrc"       %% "reactivemongo-test-26"   % "0.3.0"          % Test,
```
Comment (konradwudkowski) [inline build.sbt]: it's a bit pedantic, I know but % Test could be aligned vertically

File: src/main/scala/uk/gov/hmrc/workitem/WorkItemRepository.scala [hunk 2]
```diff
@@ -38,7 +32,7 @@ abstract class WorkItemRepository[T, ID](
   collectionName: String,
   mongo: () => DB,
   itemFormat: Format[WorkItem[T]],
-  configValue: Option[Long]
+  retryAfterSeconds: Option[Long]
```
Comment (konradwudkowski) [inline src/main/scala/uk/gov/hmrc/workitem/WorkItemRepository.scala]: Have you checked how is this param usable for client code? This is an abstract class, so another class extends it. At this level that other class will usually have access to injected Configuration but not yet to the individual config value. I'm not sure

File: src/main/scala/uk/gov/hmrc/workitem/WorkItemRepository.scala [hunk 3]
```diff
@@ -63,7 +57,7 @@ abstract class WorkItemRepository[T, ID](
 
   lazy val inProgressRetryAfter: Duration = {
     Duration.millis(
-      configValue.getOrElse(
+      retryAfterSeconds.getOrElse(
```
Comment (konradwudkowski) [inline src/main/scala/uk/gov/hmrc/workitem/WorkItemRepository.scala]: So we pass an Option[Long] and in exception complain about inProgressRetryAfterProperty which had nothing to do with each other. Method inProgressRetryAfterProperty has no reason to exist if we inject retryAfter directly.

I'm thinking that maybe it's easier to inject typesafe config and keep cheking in it as before. What do you think?

File: src/main/scala/uk/gov/hmrc/workitem/WorkItemRepository.scala [hunk 4]
```diff
@@ -134,26 +128,18 @@ abstract class WorkItemRepository[T, ID](
     id.map(_.map(getWorkItem)).flatMap(_.getOrElse(Future.successful(None)))
   }
 
-  val runner = Command.run(BSONSerializationPack, FailoverStrategy.default)
```
Comment (konradwudkowski) [inline src/main/scala/uk/gov/hmrc/workitem/WorkItemRepository.scala]: I remember that removing the `runner` required change in a unit test here: WorkItemRepositorySpec.scala
we did it already on a branch here: https://github.com/hmrc/work-item-repo/tree/feature/PLATOPS-1484_play26

File: src/main/scala/uk/gov/hmrc/workitem/WorkItemRepository.scala (hunk 3 above)
Comment (konradwudkowski) [reply src/main/scala/uk/gov/hmrc/workitem/WorkItemRepository.scala]: the long value is so far defined as a Duration so it's not good to require long, but I like typesafe config option more I think

File: src/main/scala/uk/gov/hmrc/workitem/WorkItemRepository.scala (hunk 2 above)
Comment (alexflav23) [reply src/main/scala/uk/gov/hmrc/workitem/WorkItemRepository.scala]: @konradwudkowski I've only followed up with a potentially small segment of the users of this library. As a client you need to provide this value manually after reading it from an actual configuration.

**Pull request closed without merging**